src/ufo.rs

```diff
@@ -243,6 +331,42 @@ impl Ufo {
     }
 }
 
+/// Validate the contents of the groups.plist file according to the rules in the
+/// [Unified Font Object v3 specification for groups.plist](http://unifiedfontobject.org/versions/ufo3/groups.plist/#specification).
+fn validate_groups(groups_map: &BTreeMap<String, Vec<String>>) -> Result<(), Error> {
+    let mut kern1_set = HashSet::new();
+    let mut kern2_set = HashSet::new();
+    for (group_name, group_glyph_names) in groups_map {
+        if group_name.is_empty() {
+            return Err(Error::GroupsError);
+        }
+
+        if group_name.starts_with("public.kern1.") {
+            if group_name.len() == 13 {
+                // Prefix but no actual name.
+                return Err(Error::GroupsError);
+            }
+            for glyph_name in group_glyph_names {
+                if !kern1_set.insert(glyph_name) {
+                    return Err(Error::GroupsError);
+                }
+            }
+        } else if group_name.starts_with("public.kern2.") {
+            if group_name.len() == 13 {
+                // Prefix but no actual name.
+                return Err(Error::GroupsError);
+            }
+            for glyph_name in group_glyph_names {
+                if !kern2_set.insert(glyph_name) {
+                    return Err(Error::GroupsError);
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
```

Comment:
I would suggest adding a separate `GroupsError` type, that might look something like,

```rust
enum GroupsError {
    InvalidName,
    DuplicateGlyph {
        glyph_name: String,
        group_name: String,
    },
    OverlappingGroups {
        overlapping_glyphs: Vec<String>,
    },
}
```

and then you can implement `Display` for this error in a way that conveys more specific information about the error.